src/chain/chain_state.cpp

```diff
@@ -569,13 +594,20 @@ uint32_t chain_state::easy_work_required(const data& values, bool daa_active)
     auto height = values.height;
     auto& bits = values.bits.ordered;
 
-    if(daa_active) 
+    if (daa_active) { 
+#ifdef BITPRIM_CURRENCY_BCH
         return cash_difficulty_adjustment(values);
-    else
+#else
+//Note: Could not happend: DAA and not BCH
```

Comment:
This is quite a big source file; in my opinion, this is where moving the currency-specific logic to a separate file would reduce complexity and improve readability and navigability